File: scala/src/main/scala/com/xebia/functional/embeddings/openai/models/RequestConfig.scala
package com.xebia.functional.scala.embeddings.openai.models

final case class RequestConfig(model: EmbeddingModel, user: RequestConfig.User)

object RequestConfig:
  opaque type User = String
  extension (r: User) def asString: String = r

  object User:
    def apply(v: String): User = v

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/models/embeddings/RequestConfig.kt
package com.xebia.functional.xef.llm.models.embeddings

import kotlin.jvm.JvmInline
import kotlin.jvm.JvmStatic

data class RequestConfig(val user: User) {
  companion object {
    @JvmStatic
    fun apply(userId: String): RequestConfig {
      return RequestConfig(User(userId))
    }

    @JvmInline value class User(val id: String)
  }
}
